### PR TITLE
feat: Add target_model to support multi-model endpoints

### PR DESCRIPTION
**Pull request by keerthanvasist**
*Issue #, if available:*
Add `target_model` to support multi-model endpoints in Clarify processing jobs. 

*Testing done:*
Added unit tests. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/clarify.py [hunk 1]
```diff
@@ -185,6 +185,7 @@ def __init__(
         custom_attributes=None,
         accelerator_type=None,
         endpoint_name_prefix=None,
+        target_model=None,
```
[the remark this answers is not in the document]
Comment (keerthanvasist) [reply src/sagemaker/clarify.py:188]: No

File: src/sagemaker/clarify.py [hunk 2]
```diff
@@ -218,6 +219,8 @@ def __init__(
                 https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html.
             endpoint_name_prefix (str): The endpoint name prefix of a new endpoint. Must follow
                 pattern "^[a-zA-Z0-9](-\*[a-zA-Z0-9]".
+            target_model (str): Sets the target model name when using a multi-model endpoint. For
+            more information about multi-model endpoints
```
Comment (goswamig) [inline src/sagemaker/clarify.py:223]: I think you want to add link of multi-model endpoints.

**Review by goswamig — CHANGES_REQUESTED** [review]
I think you want to add link of multi-model endpoints.

**Review by goswamig — APPROVED** [review]
LGTM

**Review by Satish615 — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**